### PR TITLE
Fixes #290 Extending logfile generation to deployment

### LINKED ISSUE #290: Extending logfile generation to deployment
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

The LOGFILE is generated only for the logs of docs generation script.
### Expected behaviour

Extend this feature to the logs of Github deployment. This feature is needed as it becomes difficult to understand issues in those scripts without getting detailed logs for them.

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

* Created separate events for retrieving logs file of various scripts
* As @pri22296 suggested (https://github.com/fossasia/yaydoc/issues/290#issuecomment-316995862) removed quite mode from commands 
* Updated routes for downloading log files
* Added Detailed logs button in `deploy/github` and `deploy/heroku`
* Added modal to preview logs
* Update documentation of `socket-github.js` and `socket-heroku.js`
## Related Issue

issue: #290 

## How Has This Been Tested?

Test at https://yaydoc6.herokuapp.com

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296  @sch00lb0y  Please review.

File: ghpages_deploy.sh [hunk 1]
```diff
@@ -10,61 +10,73 @@ do
...
-  i) UNIQUE_ID=${OPTARG};;
+  i) UNIQUEID=${OPTARG};;
   n) USERNAME=${OPTARG};;
   t) OAUTH_TOKEN=${OPTARG};;
   r) REPONAME=${OPTARG};;
   o) OWNER=${OPTARG};;
   esac
 done
 
+function print_log {
```
Comment (pri22296) [inline ghpages_deploy.sh]: Define these in a new file logging.sh and source it from other scripts.

File: ghpages_deploy.sh [hunk 2]
```diff
@@ -10,61 +10,73 @@ do
...
 # Overwrite files in the branch
+print_log "Overwriting previous content with new changes!\n"
 shopt -s extglob
 rm -rf ./!(.git)
 
-cp -a ../${UNIQUE_ID}_preview/. ./
-
-echo -e "Overwrite successfully \n"
+cp -a ../${UNIQUEID}_preview/. ./
+print_log "Overwrite successfully \n"
```
Comment (pri22296) [inline ghpages_deploy.sh]: Overwrite successful

**Comment by imujjwal96** [comment]
@pri22296  Updated!

File: app.js [hunk 3]
```diff
@@ -75,10 +77,18 @@ io.on('connection', function(socket){
...
+  socket.on('retrieve-generate-logs', function (data) {
+    output.retrieveLogs(socket, 'generate', data);
+  });
+
+  socket.on('retrieve-ghpages-deploy-logs', function (data) {
+    output.retrieveLogs(socket, 'ghpages_deploy', data);
   });
 
+  socket.on('retrieve-heroku-deploy-logs', function (data) {
+    console.log("inside app.js");
```
Comment (pri22296) [inline app.js]: Is this needed?

Comment (imujjwal96) [inline app.js]: oh! hahaha

Comment (imujjwal96) [inline app.js]: updated

File: generate.sh [hunk 4]
```diff
@@ -16,13 +16,7 @@ do
...
 done
 
-function print_log {
-  echo -e $1 | tee -a ${LOGFILE}
-}
-
-function print_danger {
-  >&2 echo -e $1 | tee -a ${LOGFILE}
-}
+source logging.sh
```
Comment (pri22296) [inline generate.sh]: Maybe source it at the top of the file. It's kind of like an import.

File: generate.sh [hunk 5]
```diff
@@ -109,7 +103,7 @@ mkdir -p _themes
 
 # Setting up documentation sources
 print_log "Setting up documentation sources\n"
-sphinx-quickstart -q -v "$VERSION" -a "$AUTHOR" -p "$PROJECTNAME" -t templates/ -d html_theme=$DOCTHEME -d html_logo=$LOGO -d root_dir=$ROOT_DIR -d autoapi_python=$AUTOAPI_PYTHON -d mock_modules=$MOCK_MODULES >> ${LOGFILE}
+sphinx-quickstart -q -v "$VERSION" -a "$AUTHOR" -p "$PROJECTNAME" -t templates/ -d html_theme=$DOCTHEME -d html_logo=$LOGO -d root_dir=$ROOT_DIR -d autoapi_python=$AUTOAPI_PYTHON -d mock_modules=$MOCK_MODULES >>${LOGFILE} 2>>${LOGFILE}
```
Comment (pri22296) [inline generate.sh]: Remove this `-q`?

Comment (imujjwal96) [inline generate.sh]: I tried removing this, but then the script hanged at this location. I think maybe it asks for some user input when we remove `-q`

File: generate.sh (hunk 4 above)
Comment (imujjwal96) [inline generate.sh]: Hmm! :+1:

File: generate.sh (hunk 5 above)
Comment (pri22296) [inline generate.sh]: Ah, right. I forgot that `-q` is required for `sphinx-quickstart` for non interactive mode. Ignore this.

File: generate.sh (hunk 4 above)
Comment (imujjwal96) [inline generate.sh]: Updated!

**Pull request merged**